opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
I don't know if the compiler is that smart (we'd have to try I guess). But I have to say I'm more attached to constness by the information it brings to the developer, and the additional safety to the type system. I wouldn't personally start fiddling with const in the atomspace API however, certainly not at my current level of understanding.